api/src/main/java/com/redhat/ipaas/api/v1/rest/DataManager.java

```diff
@@ -182,74 +194,106 @@ public String generatePK(Map<String, WithId> entityMap) {
             default:
                 break;
         }
-        throw new ClassNotFoundException("No class found for model " + model);
+        throw IPaasServerException.launderThrowable(new ClassNotFoundException("No class found for model " + kind));
```

Comment:
From https://docs.oracle.com/javase/7/docs/api/java/lang/ClassNotFoundException.html, `ClassNotFoundException` probably isn't the right thing to throw here. How about `IllegalArgumentException` here too?